mat64/dense_arithmetic.go

```diff
@@ -124,12 +124,21 @@ func (m *Dense) Norm(ord float64) float64 {
 func (m *Dense) Add(a, b Matrix) {
 	ar, ac := a.Dims()
 	br, bc := b.Dims()
-
 	if ar != br || ac != bc {
 		panic(ErrShape)
 	}
 
+	aMat, _ := untranspose(a)
+	bMat, _ := untranspose(b)
 	m.reuseAs(ar, ac)
+	var f func()
+	m, f = uniqueMemoryDense(m, aMat)
+	if f == nil {
```

Comment:
Ah. Found the error. I blindly copied the code above, but the comparison should be with aMat and bMat, not a and b.